src/phase0/state_transition/epoch_processing.rs

```diff
@@ -154,8 +156,63 @@ pub fn process_registry_updates<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    // Process activation eligibility and ejections
+    let current_epoch = get_current_epoch(state, context);
+
+    for i in 0..state.validators.len() {
+        let validator = &mut state.validators[i];
+        if is_eligible_for_activation_queue(validator, context) {
+            validator.activation_eligibility_epoch = current_epoch + 1;
+        }
+
+        if is_active_validator(validator, current_epoch)
+            && validator.effective_balance <= context.ejection_balance
+        {
+            initiate_validator_exit(state, i, context);
+        }
+    }
+
+    // Queue validators eligible for activation and not yet dequeued for activation
+    let _activation_queue = state
+        .validators
+        .iter()
+        .enumerate()
+        .filter(|&(_, validator)| is_eligible_for_activation(state, validator))
+        .collect::<Vec<(ValidatorIndex, &Validator)>>()
+        // Order by the sequence of activation_eligibility_epoch setting and then index
+        .sort_by(|a, b| {
+            let a_activation_eligibility_epoch = a.1.activation_eligibility_epoch;
+            let a_index = a.0;
+            let b_activation_eligibility_epoch = b.1.activation_eligibility_epoch;
+            let b_index = b.0;
+            match a_activation_eligibility_epoch.cmp(&b_activation_eligibility_epoch) {
+                cmp::Ordering::Less => cmp::Ordering::Less,
```

Comment:
random aside but when i run into needing to match on the variants of `Ordering` i just import `std::cmp::Ordering` so that you can drop all of these `cmp`

Reply:
Kk yeah good call